delta-lake/src/main/java/org/apache/iceberg/delta/BaseSnapshotDeltaLakeTableAction.java

```diff
@@ -412,6 +426,25 @@ private Map<String, String> destTableProperties(
     return additionalPropertiesBuilder.build();
   }
 
+  private void tagCurrentSnapshot(long deltaVersion, Transaction transaction) {
+    long currentSnapshotId = transaction.table().currentSnapshot().snapshotId();
+
+    ManageSnapshots manageSnapshots = transaction.manageSnapshots();
+    manageSnapshots.createTag(DELTA_VERSION_TAG_PREFIX + deltaVersion, currentSnapshotId);
+
+    Timestamp deltaVersionTimestamp = deltaLog.getCommitInfoAt(deltaVersion).getTimestamp();
+    if (deltaVersionTimestamp != null) {
+      String formattedDeltaTimestamp =
+          deltaVersionTimestamp
+              .toInstant()
+              .atZone(ZoneId.of(DELTA_TIME_STAMP_ZONE))
+              .format(DateTimeFormatter.ofPattern(DELTA_TIME_STAMP_FORMAT));
```

Comment:
I am wondering how much it is worth doing this conversion, instead of just using the millisecond timestamp value. Any thoughts?

Reply:
Thought about this a bit more, I think I would +1 for just using raw millisecond timestamp. Dealing with timezone is complex and should be avoided if there is no significant benefit. Also it leaves a space character in the tag, that might cause issue.